package/src/components/Message/MessageSimple/MessagePinned.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { Pin } from 'stream-chat-react-native';
+
+import {
+  MessageContextValue,
+  useMessageContext,
+} from '../../../contexts/messageContext/MessageContext';
+import { useTheme } from '../../../contexts/themeContext/ThemeContext';
+
+const styles = StyleSheet.create({
+  view: {
+    display: 'flex',
+    flexDirection: 'row',
+    marginBottom: 8,
+    marginTop: 5,
+    marginLeft: 30,
+  },
+  text: {
+    marginLeft: 5,
+  },
+});
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type MessagePinnedPropsWithContext<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType,
+> = Pick<
+  MessageContextValue<At, Ch, Co, Ev, Me, Re, Us>,
+  'alignment' | 'lastGroupMessage' | 'message' | 'showAvatar'
+>;
+
+const MessagePinnedWithContext = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType,
+>(
+  props: MessagePinnedPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { alignment } = props;
+  const {
+    theme: {
+      colors: { grey },
+    },
+  } = useTheme();
+
+  return (
+    <View
+      style={[
+        styles.view,
+        {
+          justifyContent: alignment === 'left' ? 'flex-start' : 'flex-end',
+        },
+      ]}
+      testID='message-avatar'
+    >
+      <Pin height={16} width={24} pathFill={grey} />
+      <Text style={[{ color: grey }, styles.text]}>Pinned by You</Text>
```

Comment:
What if the pin is made by somebody else than me? :D Do we store who the pin was made by? If not, the wording doesn't seem to be the best imho

Reply:
I am not sure if that is the case too. I will have to check if this is going to be for other users too.